chain/cmd/bandsv/main.go

```diff
@@ -15,7 +17,61 @@ type OracleRequest struct {
 type OracleRequestResp struct {
 	RequestId uint64 `json:"id"`
 	CodeHash  []byte `json:"codeHash"`
+}
+
+type GetRequestResp struct {
+	Request zoracle.RequestWithReport `json:"request"`
+	Proof   Proof                     `json:"proof"`
+}
+
+type IAVLPath struct {
+	SubtreeHeight  uint64 `json:"subtreeHeight"`
+	SubtreeSize    uint64 `json:"subtreeSize"`
+	SubtreeVersion uint64 `json:"subtreeVersion"`
+	IsDataOnRight  bool   `json:"isDataOnRight"`
+	SiblingHash    []byte `json:"siblingHash"`
+}
+
+type LeafNode struct {
+	Version   uint64 `json:"version"`
+	RequestId uint64 `json:"requestId"`
+	CodeHash  []byte `json:"codeHash"`
 	Params    []byte `json:"params"`
+	Value     []byte `json:"value"`
+}
+
+type BlockHeaderMerkleParts struct {
+	VersionAndChainIdHash       []byte `json:"versionAndChainIdHash"`
+	TimeHash                    []byte `json:"timeHash"`
+	TxCountAndLastBlockInfoHash []byte `json:"txCountAndLastBlockInfoHash"`
+	ConsensusDataHash           []byte `json:"consensusDataHash"`
+	LastResultsHash             []byte `json:"lastResultsHash"`
+	EvidenceAndProposerHash     []byte `json:"evidenceAndProposerHash"`
+}
+
+type BlockHashProof struct {
+	ZoracleRootHash        []byte                 `json:"zoracleRootHash"`
+	OtherStoreHash         []byte                 `json:"otherStoreHash"`
+	BlockHeaderMerkleParts BlockHeaderMerkleParts `json:"blockHeaderMerkleParts"`
+	SignedDataPrefix       []byte                 `json:"signedDataPrefix"`
+	Signatures             []TMSignature          `json:"signatures"`
+}
+
+type ZoracleTreeProof struct {
+	LeafNode LeafNode   `json:"valueNode"`
+	Paths    []IAVLPath `json:"paths"`
+}
+type TMSignature struct {
+	R                []byte `json:"r"`
+	S                []byte `json:"s"`
+	V                uint8  `json:"v"`
+	SignedDataSuffix []byte `json:"signedDataSuffix"`
+}
+
+type Proof struct {
+	BlockHeight      uint64           `json:"blockHeight"`
+	ZoracleTreeProof ZoracleTreeProof `json:"zoracleTreeProof"`
+	BlockHashProof   BlockHashProof   `json:"blockHashProof"`
```

Comment:
Is it ok to have field name the same as struct name?

Reply:
ok